### ISSUE TITLE
LaggersDetect 

### ISSUE THREAD

**Issue by v0UwUK6CmCrjVB1PAtRb** [RogueDrifter/Anti_cheat_pack#20]
I have a shop system which allows players to purchase weapons via a ShowPlayerDialog. The dialog appears just fine, but when hitting "Select" to confirm the purchase, nothing happens, the dialog will close (as expected), but no weapon will be given to the player. 

If I comment/disable `#include "LaggersDetect.inc"` in  my gamemode, everything is fine, so it appears that somehow, this include is causing problems.

Any feedback would be greatly appreciated.

**Comment by RogueDrifter** [comment]
Okay so, Laggers detect include does not interact with dialogs at all, 

The only thing that COULD cause this, is the hooking of 'GivePlayerWeapon' function. 

And i know that i did no mistakes with it because i'm using it myself alongside a lot of many other people and the function works fine (Plus i just reviewed the code for the sake of sanity checks).

Which means that one of your includes in the script that has the dialog you mentioned (Or the script itself) has a wrong hooking for the GivePlayerWeapon function or maybe your a_samp includes are deprecated, so can you look into that and get back to me with feedback? And could you also tell me is the script that has this Dialog a filterscript or a gamemode? And what sa-mp version are you running?

**Comment by v0UwUK6CmCrjVB1PAtRb** [comment]
Thanks for the response.

Interesting theory. I've got another command that also uses GivePlayerWeapon, and that works fine. 

In both cases, the commands are in the actual gamemode (not a filterscript). I'm running server 0.3.7-R2.

Here is an example of a dialog/GivePlayerWeapon function that works fine: https://pastebin.com/FLEHn2ai

And, an example where nothing happens when hitting select in the dialog: https://pastebin.com/RqdDeFgh

They look pretty similar to me, but perhaps I'm missing something obvious...

**Comment by RogueDrifter** [comment]
```
ShowWeaponPurchaseDialog(playerid);
```

What is this function? Why not just a normal ShowPlayerDialog like the other one?

Also does this ever show when the dialog disappears?
```
GameTextForPlayer(playerid, "~n~~r~~h~Weapon purchased!", 3000, 3);
```

**Comment by v0UwUK6CmCrjVB1PAtRb** [comment]
ShowWeaponPurchaseDialog is the entire weapon shop dialog ( https://pastebin.com/R8Y9nF51 ), the other functions use a weapon specific version.

The GameTextForPlayer doesn't show up when the dialog disappears.

Is there any debugging I can add for LaggersDetect.inc to try and help figure out what's going on here?

**Comment by v0UwUK6CmCrjVB1PAtRb** [comment]
Sorry for a double comment, I just wanted to add:

I just tried the old version of OPBA.inc and FLaggers.inc and my /shop command works fine. I used this include for months with no issues.

So, it appears that something in the new version of FLaggers.inc (now LaggersDetect.inc) has caused this problem.

I'm open to using the old version, but would of course prefer to use the latest updates if possible.
